### ISSUE TITLE
Open API Media Type drop down values

### ISSUE THREAD

**Issue by jduffy3** [http4k/http4k#626]
Looking at this example https://www.http4k.org/blog/documenting_apis_with_openapi/

If I update the returning method to be
`returning(OK, Body.string(TEXT_PLAIN).toLens() to "Here's a drink", "A successful offer of a drink to the lowly meatbag.")`

The spec generated still shows 

```
      responses:
        '200':
          description: A successful offer of a drink to the lowly meatbag.
          content: {}
```
However I would've expected the following

```      
responses:
        '200':
          description: A successful offer of a drink to the lowly meatbag.
          content: 
             text/plain:
                example: "Here's a drink"
```

Is there a reason it gets blanked out?

**Comment by daviddenton** [comment]
This is happening because the Http4k open API rendering only really currently supports JSON as an output format.

**Comment by jduffy3** [comment]
Apologies, the swagger online editor converted it from JSON to YAML.

The output format is JSON but it doesn't fill out the spec when other Content-Types e.g. PLAIN_TEXT are specified.

I would've imagined based on the lens provided to the contract route that it would know how to render it in the swagger spec? Also since the Lens can take the ContentType as an argument.

e.g

```
Body.auto<String>("Description", Strict, TEXT_PLAIN).toLens()
```

**Comment by jduffy3** [comment]
Are there any examples of handling Content Negotiation with a swagger spec?

**Comment by daviddenton** [comment]
@jduffy3 Have made your changes, modified the tests slightly to add non-JSON examples.... and all the tests passed ok (with expected changes). I also added support for request bodies which your changeset didn't cover. Hopefully there will be no other effects. :) 

This will be released with 4.9.9.0

**Comment by jduffy3** [comment]
Very cool! Thanks @daviddenton ! :)
